ls8/cpu.c

```diff
@@ -172,26 +173,43 @@ void cpu_run(struct cpu *cpu)
       SP++; //from POP
       break;
 
-      // case JMP: //Jump to the address stored in the given register.
-      // cpu->PC = cpu->reg[operandA]; //1. Jump to the address stored in the given register.Set the PC to the address stored in the given register.
-      // break;
+      case CMP: //This is an instruction handled by the ALU.
+      alu(cpu, ALU_CMP, operandA, operandB);
+      cpu->PC += 3;
+      break;
+
+      case JMP: //Jump to the address stored in the given register.
+      cpu->PC = cpu->reg[operandA]; //Set the PC to the address stored in the given register.
+      break;
+
+      case JEQ:
+      if (cpu->FL == 0b00000001) //if equal flag is set (true), 
+      {
+        cpu->PC = cpu->reg[operandA]; //jump to the address stored in the given register.
+      }
+      cpu->PC +=2;//if it doesn't jump
+      break;
 
-      // case CMP: //This is an instruction handled by the ALU.
-      // alu(cpu, ALU_CMP, operandA, operandB);
-      // break;
+      case JNE:
+      if (cpu->FL == 0b00000000)//If E flag is clear (false, 0)
```

Comment:
This logic is slightly off, you want the case where "cpu->FL is not equal to the value that indicates it is true" I know that is a cryptic way of saying it, but I don't want to give it away too easily if you want to think about how to fix it. It's a pretty simple fix.